automata/base/animation.py

```diff
@@ -0,0 +1,255 @@
+from dataclasses import dataclass
+from typing import Iterable, Optional
+
+import manim
+import pygraphviz as pgv
+
+
+class Animate:
+    """The `Animate` class is the config class to set the behavior of animations."""
+
+    DEFAULT_COLOR = manim.WHITE
+    HIGHLIGHT_COLOR = manim.RED
+    manim.Circle.set_default(color=DEFAULT_COLOR)
```

Comment:
Does this change a global property of `manim`? Is there a way to do this local to what is animated from this class?

Reply:
Yes, it changes the default color of `manim.Circle` from `manim.RED` to `manim.WHITE` globally. A way to do this local to what is animated from this class is by specifying `color=Animate.DEFAULT_COLOR` in every `manim.MObject` created, which is what I did at first. After that, I thought it may be too troublesome. As most of `manim.MObject` except `manim.Circle` uses `manim.WHITE` as default color, I chose to change the default color of `manim.Circle` to `manim.WHITE` globally.

Is the way of specifying `color=Animate.DEFAULT_COLOR` in every `manim.MObject` created suggested? If so, I'll revise the code in that way.

Reply:
Is a helper function like this OK? (`partial` is from `functools`)
```suggestion
    M = TypeVar("M", bound=manim.Mobject)

    @classmethod
    def default_init(cls, mobject_class: type[M]):
        return partial(mobject_class, color=cls.DEFAULT_COLOR)
```
It can handle all subclasses of `manim.Mobject` like `manim.Circle`, `manim.Text`, etc. In that case, we would use `Animate.default_init(/*a manim.Mobject class*/)` to replace the default init function of that class. For example, `manim.Text(node.name, font_size=float(node.attr["fontsize"]))` will be replaced by `Animate.default_init(manim.Text)(node.name, font_size=float(node.attr["fontsize"]))`, and `manim.Circle()` will be repalced by `Animate.default_init(manim.Circle)()` (seems that the latter `()` are easy to forget though).

Reply:
I think that seems reasonable! My main motivation for this comment is to avoid side effects that can cause other animation code to behave unexpectedly, and I think this is a solid way of doing that.